src/rpc/server.rs

```diff
@@ -74,24 +74,29 @@ pub async fn receive<H: Handler + std::marker::Sync + std::marker::Send + 'stati
 ) {
     loop {
         {
-            debug!("parse_request, start");
+            let id = connection.id();
+            debug!("{} parse_request, start", id);
             let header = match connection.receive_request_header(&mut read_stream).await {
                 Ok(header) => header,
                 Err(e) => {
-                    debug!("parse_request, header error: {}", e);
+                    if e.to_string() == "early eof" {
+                        warn!("connection {} is closed abnormally.", id);
```

Comment:
#59 bug fixed here

Reply:
It's better to split two PRs. And printing a message isn't enough.